src/transformers/file_utils.py

```diff
@@ -298,6 +298,43 @@ def is_torch_cuda_available():
         return False
 
 
+def is_torch_bf16_available():
+    if is_torch_available():
+        import torch
+
+        # XXX: copying test function from https://github.com/pytorch/pytorch/blob/2289a12f21c54da93bf5d696e3f9aea83dd9c10d/torch/testing/_internal/common_cuda.py#L51 until utility function is made available
+        # with additional check for torch version
+        # to succeed:
+        # 1. the hardware needs to support bf16 (arch >= Ampere)
+        # 2. torch >= 1.9
+        # 3. CUDA >= 11
+        # XXX: one problem here is that it may give invalid results on mixed hardware, so it's
+        # really only correct for the 0th gpu (or currently set default device if different from 0)
+
+        if not torch.cuda.is_available() or torch.version.cuda is None:
+            return False
+        if torch.cuda.get_device_properties(torch.cuda.current_device()).major < 8:
+            return False
+        if int(torch.version.cuda.split(".")[0]) < 11:
+            return False
+        if not version.parse(torch.__version__) >= version.parse("1.09"):
+            return False
+
+        return True
+    else:
+        return False
+
+
+def is_torch_bf16_autocast_available():
```

Comment:
Those two functions should be merged in just one as the use case of using bfloat16 for just inference seems very narrow to me.

Reply:
So you propose to have just `is_torch_bf16_autocast_available()`, correct?

Reply:
I would name it `is_torch_bf16_available()` for the sake of simplicity, but yes, it would return False if bf16 is available but not in amp.autocast.

Reply:
But bf16-support has been available since 1.9.0, so that would result in a misleading function, should someone want to deploy it with pt-1.9.0 and will be puzzled that it'd return false since autocast/fp16 won't be there.

Unless perhaps you meant something else, when you said to merge the 2.

Reply:
I disagree that support for bf16 has been available since 1.9.0 since it does not support what we need for training. You are talking about a very narrow edge case that can easily be documented.

Reply:
Not even `self.args.half_precision_full_eval == bf16`? (using the newly proposed cli arg)

Reply:
In that specific case, the user would just need to upgrade to PyTorch 1.10, which is going to be released before we merge this anyway. Not sure it's worth all the extra logic.

It would be different if the autocast addition required something more annoying like a more recent cuda version, but I think for this we can just say bfloat16 support will require PyTorch 1.10+, no?

Reply:
> It would be different if the autocast addition required something more annoying like a more recent cuda version

It does require cuda-11

```
if int(torch.version.cuda.split(".")[0]) < 11:
```

But I'm fine with your proposal.

Reply:
This is required by the basic bfloat16 support, it's not added by the autocast addition.

Reply:
done